File: src/main/java/models/desktop/landingPage/LandingPage.java
package models.desktop.landingPage;

import models.desktop.landingPage.searchBarCars.SearchBarCars;
import models.desktop.landingPage.searchBarFlights.SearchBarFlights;
import models.desktop.landingPage.searchBarHotels.SearchBarHotels;
import models.desktop.landingPage.searchBarTours.SearchBarTours;
import models.desktop.landingPage.searchBarVisa.SearchBarVisa;

import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
import static utils.logger.Log4J.log;

public class LandingPage extends LandingPageLocators {
    public static ThreadLocal<Boolean> IS_AD_CLOSED = ThreadLocal.withInitial(() -> false);

    public LandingPage() {
        browser.waitForPageLoaded(30);
        checkUrl();
        hideAd();
        log.info("Landing page has been displayed.");
    }

    public SearchBarFlights clickOnFlightsTab() {
        click.clickOnElement(tabFlights, 15);
        log.info("Flights tab has been clicked.");

        return new SearchBarFlights();
    }

    public SearchBarHotels clickOnHotelsTab() {
        click.clickOnElement(tabHotels, 15);
        log.info("Hotels tab has been clicked.");

        return new SearchBarHotels();
    }

    public SearchBarTours clickOnToursTab() {
        click.clickOnElement(tabTours, 15);
        log.info("Tours tab has been clicked.");

        return new SearchBarTours();
    }

    public SearchBarCars clickOnCarsTab() {
        click.clickOnElement(tabCars, 15);
        log.info("Cars tab has been clicked.");

        return new SearchBarCars();
    }

    public SearchBarVisa clickOnVisaTab() {
        click.clickOnElement(tabVisa, 15);
        log.info("Visa tab has been clicked.");

        return new SearchBarVisa();
    }

    private void checkUrl() {
        String expectedUrl = "https://phptravels.net/";
        String actualUrl = get.getCurrentUrl();

        assertThat(actualUrl)
                .as("URL check")
                .isEqualTo(expectedUrl);
    }

    private void hideAd() {
        if (!LandingPage.IS_AD_CLOSED.get()) {
            if (check.isElementDisplayed(hideAdButton, 10)) {
                click.clickOnVisibleElement(hideAdButton, 10);
                log.info("Hide ad button has been clicked.");

                LandingPage.IS_AD_CLOSED.set(true);
            }
        }
    }
}

File: ts-playwright-test-automation-framework/src/main/code/models/web/landingPage/LandingPage.ts
import { LandingPageSelectors } from "./LandingPageSelectors";
import { SearchBarCars } from "./searchBarCars/SearchBarCars";
import { SearchBarFlights } from "./searchBarFlights/SearchBarFlights";
import { SearchBarHotels } from "./searchBarHotels/SearchBarHotels";
import { SearchBarTours } from "./searchBarTours/SearchBarTours";
import { SearchBarVisa } from "./searchBarVisa/SearchBarVisa";
import { logger } from "../../../utils/logger/Logger";

export class LandingPage extends LandingPageSelectors {
    private log = logger.child({ label: LandingPage.name });

    public async checkIfPageIsLoaded(): Promise<LandingPage> {
        await this.checkUrl();
        await this.check.isVisible(this.background, "Background")
        this.log.info("Landing page has been displayed.");

        return this;
    }

    public async clickOnFlightsTab(): Promise<SearchBarFlights> {
        await this.click.leftClick(this.tabFlightsSelector, "Flights tab");
        this.log.info("Flights tab has been clicked.");

        return new SearchBarFlights();
    }

    public async clickOnHotelsTab(): Promise<SearchBarHotels> {
        await this.click.leftClick(this.tabHotelsSelector, "Hotels tab");
        this.log.info("Hotels tab has been clicked.");

        return new SearchBarHotels();
    }

    public async clickOnToursTab(): Promise<SearchBarTours> {
        await this.click.leftClick(this.tabToursSelector, "Tours tab");
        this.log.info("Tours tab has been clicked.");

        return new SearchBarTours();
    }

    public async clickOnCarsTab(): Promise<SearchBarCars> {
        await this.click.leftClick(this.tabCarsSelector, "Cars tab");
        this.log.info("Cars tab has been clicked.");

        return new SearchBarCars();
    }

    public async clickOnVisaTab(): Promise<SearchBarVisa> {
        await this.click.leftClick(this.tabVisaSelector, "Visa tab");
        this.log.info("Visa tab has been clicked.");

        return new SearchBarVisa();
    }

    private async checkUrl(): Promise<void> {
        let expectedUrl: string = "https://phptravels.net/";
        await this.check.doesUrlContain(expectedUrl);
    }
}